Dockerfile

```diff
@@ -1,94 +1,115 @@
+        ca-certificates \
+        libsqlite3-0 \
+        libssl1.0.0 \
+        libgfortran3 \
+        git \
+        postgresql \
+        postgresql-contrib \
+        libpq-dev && \
+    rm -rf /var/lib/apt/lists/*
+
+# We like PGP and so are going through this pain. Notes for the future. Yay!
+# I *think* this is how this works.
 #
-# Note that the python version is also listed in the enviornment.yml
-# file. The version in CIVIS_PYTHON_VERSION is the source of truth.
-# If you want to change the python version, you need to change it in
-# **both** places. The python version has been left in the `environment.yml`
-# file so that people can create environments equivalent to this
-# container. However, change the name of the environment first!
+#     1) Figure out who the release manager is for your python version. (I *think* they usually make
+#        a PEP for it, because, like, they make a PEP for everything. See here for 3.6.0
+#        https://www.python.org/dev/peps/pep-0494/.)
+#     2) Then go look them up on a key server (This one is nice https://pgp.mit.edu/)
+#     3) Search by their name and find the short 8 character key ID. You might have to try a few
+#        different keys since people can have more than one and it is not always clear which one
+#        was used to sign the package.
+#     4) Put the 8 character key ID below.
 #
-# The ordering of these steps seems to matter. You seem to have to
-# install a specific python version by hand and then pin it.
-# 1) install conda
-# 2) pin conda to the version given by CIVIS_CONDA_VERSION
-# 3) install the python version CIVIS_PYTHON_VERSION
-# 4) pin the python version
-RUN echo 'export PATH=/opt/conda/bin:$PATH' > /etc/profile.d/conda.sh && \
-    wget --quiet https://repo.continuum.io/miniconda/Miniconda3-${CIVIS_CONDA_VERSION}-Linux-x86_64.sh && \
-    /bin/bash /Miniconda3-${CIVIS_CONDA_VERSION}-Linux-x86_64.sh -b -p /opt/conda && \
-    rm Miniconda3-${CIVIS_CONDA_VERSION}-Linux-x86_64.sh && \
-    /opt/conda/bin/conda install --yes conda==${CIVIS_CONDA_VERSION} && \
-    echo "conda ==${CIVIS_CONDA_VERSION}" > /opt/conda/conda-meta/pinned && \
-    conda install --yes python==${CIVIS_PYTHON_VERSION} && \
-    echo "\npython ==${CIVIS_PYTHON_VERSION}" >> /opt/conda/conda-meta/pinned
+# If everything goes well, you will successfully verify that the download (over https) in fact
+# returned the correct, uncorrupted file. Yay!
+ENV GPG_KEY AA65421D
+
+ENV PYTHON_VERSION 3.6.0
+
+# If this is called "PIP_VERSION", pip explodes with "ValueError: invalid truth value '<VERSION>'"
+ENV PYTHON_PIP_VERSION 9.0.1
+
+RUN set -ex && \
+    buildDeps=' \
+        curl \
+        gcc \
+        g++ \
+        gfortran \
+        libbz2-dev \
+        libc6-dev \
+        liblzma-dev \
+        libncurses-dev \
+        libreadline-dev \
+        libsqlite3-dev \
+        libssl-dev \
+        make \
+        xz-utils \
+        zlib1g-dev' && \
```

Comment:
Why not install these with the first apt-get? It seems unnecessarily obfuscated as a string here.